server/api/src/main/java/org/hawkular/apm/server/api/utils/ProducerInfoUtil.java

```diff
@@ -196,6 +197,10 @@ public static ProducerInfo getProducerInfo(String tenantId, Span serverSpan, Spa
                 pi.getProperties().addAll(clientSpan.binaryAnnotationMapping().getProperties());
                 pi.setHostAddress(clientSpan.ipv4());
 
+                if (serverSpan.service() != null) {
```

Comment:
Should be clientSpan.service() - although it probably won't matter as the client and server spans are in the same service - but all of the other information populating the ProducerInfo is based on the clientSpan.